Fix command splitting and joining when an argument contains spaces

emillynge opened the pull request:
At the moment ECS and compose transformers performs a "dumb" argument splitting.

This means that the following compose entrypoint:
```
entrypoint: ['/bin/echo', 'Hello world']
```
gets incorrectly transformed into `/bin/echo hello world` which fails as echo only expects 1 argument.

conversely ECS emits `'/bin/echo "Hello world"` as 
 ```
"entryPoint": [
                "/bin/echo",
                "\"Hello",
                "world\""
            ]
```
This PR fixes these and adds tests for the failure mode.

micahhausler commented:
@emillynge thanks for making this! Any chance you could squash the 3 commits down to one?

emillynge commented:
I would, but it seems like a hassle that is unnecessary since this is actually built-in as a tool for the _maintainer_.

[Squash to only one “proper” commit for github pull request](https://stackoverflow.com/questions/11396069/squash-to-only-one-proper-commit-for-github-pull-request)

> Note that Bob doesn't have to squash his commits when he is making a GitHub PR.
Since March 2016, you can leave that operation to the maintainer (you) accepting your PR.

The thread refers to this post [Squash your commits](https://github.com/blog/2141-squash-your-commits)

emillynge commented:
By the way, I have expanded the code change to the kubernetes, marathon and chronos module as well for consistency.
I don't use these, so please check that I haven't fucked something up.

Also I have changed the "unsplitting" slightly as I found out that `shlex.quote()` is a bit agressive (quoting stuff that did not need escaping).
Now I use `shlex.split` to check whether quoting is necessary for each item in a list of commands. To have this, slightly more complex, code declared only once, I have taken the liberty of adding a static method `._list2cmdline(commands: list) -> str` to the Transformer base class. (I couldn't find any utils module)

Commit:
Use proper command line splitting and joining using shlex

emillynge commented:
Found the time to squash to just 1 commit. :)

Should be ready to merge without further ado

emillynge commented:
I'm thinking @micahhausler maybe didn't get a notification :)

Everything should be ready to merge.

micahhausler commented:
Thanks @emillynge !
